code/game/objects/items/nanoforgedkatana.dm

```diff
@@ -0,0 +1,244 @@
+/obj/item/nanoforgedkatana
+	name = "Nanoforged Katana"
+	desc = "Glorious space nippon steel, folded a million times, producing the finest blade known to mankind. \
+			This blade seems to transmit knowledge of how to use its powers directly to its owner."
+	icon = 'icons/obj/items_and_weapons.dmi'
+	icon_state = "nanoforgedkatana"
+	item_state = "nanoforgedkatana"
+	lefthand_file = 'icons/mob/inhands/weapons/swords_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/swords_righthand.dmi'
+	flags_1 = CONDUCT_1
+	obj_flags = UNIQUE_RENAME
+	w_class = WEIGHT_CLASS_BULKY
+	force = 30
+	throwforce = 70
+	throw_speed = 3
+	throw_range = 7
+	sharpness = IS_SHARP_ACCURATE
+	embedding = list( "pain_mult" = 0, "jostle_pain_mult" = 0, "embed_chance" = 300, "armour_block" = 100, "fall_chance" = 0, "ignore_throwspeed_threshold" = TRUE)
+	// Last ditch effort to kill someone. Obviously, terrible idea unless the situation is desperate, as it WILL embed and WILL need surgery to remove. No extra embed damage tho, so as to not make it op.
+	hitsound = 'sound/weapons/anime_slash.ogg'
+	armour_penetration = 75
+	block_level = 2
+	block_upgrade_walk = 0 // Base desword blocking, but no possible improvements.
+	block_power = 60 //lower block power.
+	block_sound = 'sound/weapons/egloves.ogg' //placeholder
+	block_flags = BLOCKING_ACTIVE | BLOCKING_PROJECTILE
+	max_integrity = 200
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 70, "stamina" = 0)
+	resistance_flags = FIRE_PROOF | ACID_PROOF
+	attack_verb = list("attacked", "slashed", "stabbed", "sliced", "diced", "cut")
+	var/linking_time = 40
+	var/feeding_time = 100
+	var/primed = FALSE // necessary for anime bullshit, determines dash attack readiness
+	var/mob/living/owner //defining an owner is, as far as I know, necessary to allow on_enter_storage to register COMSIG_CLICKON. Equipped would have done the job much better, but doesn't work when inserting into storage.
+	var/obj/item/storage/belt/nanoforgedkatana/current_sheathe //needed to prevent wacky shit happening by removing the blade after it is primed.
+	var/dash_sound = 'sound/weapons/unsheathed_blade.ogg'
+	var/beam_effect = "blood_beam"
+	var/phasein = /obj/effect/temp_visual/dir_setting/cult/phase
+	var/phaseout = /obj/effect/temp_visual/dir_setting/cult/phase
+	actions_types = list(/datum/action/item_action/nanoforgedkatana)
+
+/obj/item/nanoforgedkatana/Initialize()
+	. = ..()
+	AddComponent(/datum/component/butchering, 25, 90, 5) //Not made for scalping victims, but will work nonetheless
+
+/obj/item/nanoforgedkatana/suicide_act(mob/user)
```

Comment:
you should make this count as a kill for priming the katana

Reply:
Butchering or suicide ? In the first case, i'd not agree, because it's already covered by simplemobs switching from "live" to "dead" state in the attack action
in the second case, since the owner of the sword is dead, then you'd need to become its owner again ... meaning it'd reprime. However, this is a fair point on the fact that you could convince someone else to suicide then profit off of that. Adding it !